bridges/otelzap/core.go

```diff
@@ -0,0 +1,135 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package otelzap provides a bridge between the [go.uber.org/zap] and
+// OpenTelemetry logging.
+package otelzap // import "go.opentelemetry.io/contrib/bridges/otelzap"
+
+import (
+	"context"
+	"slices"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"go.opentelemetry.io/otel/log"
+)
+
+const (
+	bridgeName = "go.opentelemetry.io/contrib/bridge/zapcore"
+)
+
+// Core is a [zapcore.Core] that sends logging records to OpenTelemetry.
+type Core struct {
+	logger log.Logger
+	attr   []log.KeyValue
+	ctx    context.Context
+}
+
+// Compile-time check *Core implements zapcore.Core.
+var _ zapcore.Core = (*Core)(nil)
+
+// NewOTelZapCore creates a new [zapcore.Core] that can be used with zap.New()
+// this instance will translate zap logs to opentelemetry logs and export them.
+func NewOTelZapCore(opts ...Option) zapcore.Core {
+	cfg := newConfig(opts)
+	return &Core{
+		logger: cfg.logger(),
+		ctx:    context.Background(),
+	}
+}
+
+// LevelEnabler decides whether a given logging level is enabled when logging a message.
+func (o *Core) Enabled(level zapcore.Level) bool {
+	r := log.Record{}
+	r.SetSeverity(getOtelLevel(level))
+	return o.logger.Enabled(context.Background(), r)
+}
+
+// With adds structured context to the Core.
+func (o *Core) With(fields []zapcore.Field) zapcore.Core {
+	clone := o.clone()
+	ctx, attr := getAttr(clone.ctx, fields)
+	clone.ctx = ctx
+	clone.attr = append(clone.attr, attr...)
+	return clone
+}
```

Comment:
This should handle [namespaces](https://pkg.go.dev/go.uber.org/zap#Namespace). `Core.Write` also has to handle the namespaces.

The namespace handling code can be heavily inspired by groups implementation in slog bridge. See: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/b343370e70b6e5a4881b7cd9915af95bd58d121f/bridges/otelslog/handler.go